### PR TITLE
Port MVDR tutorial

### PR DESCRIPTION
**Pull request by mthrok**
Port MVDR tutorial to Sphinx-gallery.

### CODE REVIEW

File: examples/gallery/mvdr/mvdr_tutorial.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+# 
+# First, we import the necessary packages and retrieve the data.
+# 
+# The multi-channel audio example is selected from
+# `ConferencingSpeech <https://github.com/ConferencingSpeech/ConferencingSpeech2021>`__
+# dataset.
+# 
+# The original filename is
+# 
+#    ``SSB07200001\#noise-sound-bible-0038\#7.86_6.16_3.00_3.14_4.84_134.5285_191.7899_0.4735\#15217\#25.16333303751458\#0.2101221178590021.wav``
```
Comment (mthrok) [inline examples/gallery/mvdr/mvdr_tutorial.py:38]: @nateanl I want to make sure that we list all the usages of licensed materials.
Can you check that I listed all the original materials bellow?

I did not get what the remaining numbers mean, and I am guessing that they are some augmentation parameters and not another data. 

`7.86_6.16_3.00_3.14_4.84_134.5285_191.7899_0.4735`
`15217`
`25.16333303751458`
`0.2101221178590021`

File: examples/gallery/mvdr/mvdr_tutorial.py [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+# Results
+# =======
+# 
+
+######################################################################
+# Single-channel mask results
+# ---------------------------
+# 
+
+for solution in results_single:
```
Comment (mthrok) [inline examples/gallery/mvdr/mvdr_tutorial.py:209]: I feel like this is better listed with the audio sample (and maybe with spectrogram). What do you think?

File: docs/source/index.rst [hunk 3]
```diff
@@ -48,6 +48,7 @@ The :mod:`torchaudio` package consists of I/O, popular datasets and common audio
 
    auto_examples/wav2vec2/index
    auto_examples/tts/index
+   auto_examples/mvdr/index
```
Comment (mthrok) [inline docs/source/index.rst:51]: I think that the structure of examples can be improved, let me follow up on that later.

File: examples/gallery/mvdr/mvdr_tutorial.py (hunk 1 above)
Comment (nateanl) [reply examples/gallery/mvdr/mvdr_tutorial.py]: According to the [mixture method](https://github.com/ConferencingSpeech/ConferencingSpeech2021/blob/master/simulation/mix_wav.py#L330),
```python
utt_id = clean + seg + noise + seg + rir + seg + start_time + seg + snr + seg + scale + '.wav'
```
``seg`` is  "\#"
``7.86_6.16_3.00_3.14_4.84_134.5285_191.7899_0.4735`` is the rir file name, the rir is simulated by the image method.
``15217`` is the start_time
``25.16333303751458`` is the snr
``0.2101221178590021`` is the scale

File: examples/gallery/mvdr/mvdr_tutorial.py (hunk 2 above)
Comment (nateanl) [reply examples/gallery/mvdr/mvdr_tutorial.py]: Yeah, that's a good idea.

**Review by nateanl — APPROVED** [review]

**Pull request merged**